### PR TITLE
feat: allow disabling supervisors or running cloud in api only mode

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary

Adds the `--disable-all-supervisor` flag to run in API only mode.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: cmd/cloud/server.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ func init() {
 	serverCmd.PersistentFlags().Bool("deploy-minio-operator", true, "Whether to deploy the minio operator.")
 	serverCmd.PersistentFlags().Int64("default-max-schemas-per-logical-database", 10, "When importing and creating new proxy multitenant databases, this value is used for MaxInstallationsPerLogicalDatabase.")
 	serverCmd.PersistentFlags().String("provisioner", "kops", "Specifies which provisioner to use, one of: kops, eks.")
+	serverCmd.PersistentFlags().Bool("disable-all-supervisors", false, "disable all provisioners and enable API only funcionality")
```
Comment (gabrieljackson) [inline cmd/cloud/server.go:71]:
```suggestion
	serverCmd.PersistentFlags().Bool("disable-all-supervisors", false, "disable all supervisors (API-only functionality)")
```

**Review by gabrieljackson — COMMENTED** [review]
Looks good, just one request to clarify the flag description a bit.

File: cmd/cloud/server.go (hunk 1 above)
Comment (fmartingr) [reply cmd/cloud/server.go]: 😅

**Review by gabrieljackson — APPROVED** [review]

**Review by stylianosrigas — APPROVED** [review]
LGTM

**Pull request merged**